web/src/Utils.ts

```diff
@@ -48,7 +48,9 @@ export function findDValue(graphData: GraphData[], goalYValue: number, bridgeNam
     }
     return false
   })
-  if (!indexOfClosestHigherYValue) throw new Error('Failed to find D-value of bridge')
+  if (!indexOfClosestHigherYValue) {
+    return new Error('Failed to find D-value of bridge')
```

Comment:
Instead of returning an Error. A common pattern is to return -1/null/undefined etc.
-1 is nice so we dont need to check type